dolphin/drivers/dell_emc/vmax/client.py

```diff
@@ -123,3 +123,58 @@ def list_pools(self):
             LOG.error("Failed to get pool metrics from VMAX: {}".format(err))
             raise exception.StorageBackendException(
                 reason='Failed to get pool metrics from VMAX')
+
+    def list_volumes(self, storage_id):
+
+        try:
+            # List all volumes
+            volumes = self.conn.provisioning.get_volume_list(filters={'data_volume': 'false'})
+
+            volume_list = []
+            for volume in volumes:
+                # Get volume details
+                vol = self.conn.provisioning.get_volume(volume)
+
+                total_cap = vol['cap_mb'] * units.Mi
+                used_cap = (total_cap * vol['allocated_percent']) / 100.0
+                free_cap = total_cap - used_cap
+
+                # TODO: Update constants.VolumeStatus to make mapping more precise
+                switcher = {
+                    'Ready': constants.VolumeStatus.AVAILABLE,
+                    'Not Ready': constants.VolumeStatus.ERROR,
```

Comment:
Does these state necessarily mean in ERROR. Example, Write disabled is not Error. It's the setting that admin can set for write enabled volumes

Reply:
As mentioned in TODO, this is being analyzed. Currently we support either 'available' and 'error'

Reply:
Please Open an issue with proper priority to track this

Reply:
I already have issue opend for this.https://github.com/sodafoundation/SIM/issues/47